lib/widgets/compose_box.dart

```diff
@@ -1484,6 +1552,60 @@ class _ErrorBanner extends _Banner {
   }
 }
 
+class _EditMessageBanner extends _Banner {
+  const _EditMessageBanner({required this.composeBoxState});
+
+  final ComposeBoxState composeBoxState;
+
+  @override
+  String getLabel(ZulipLocalizations zulipLocalizations) =>
+    zulipLocalizations.composeBoxBannerLabelEditMessage;
+
+  @override
+  Color getLabelColor(DesignVariables designVariables) =>
+    designVariables.bannerTextIntInfo;
+
+  @override
+  Color getBackgroundColor(DesignVariables designVariables) =>
+    designVariables.bannerBgIntInfo;
+
+  @override
+  Widget buildTrailing(context) {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    return Row(mainAxisSize: MainAxisSize.min, spacing: 8, children: [
+      ZulipWebUiKitButton(label: zulipLocalizations.composeBoxBannerButtonCancel,
+        onPressed: () {
+          composeBoxState.endEditInteraction();
+        }),
+      // TODO(#1481) disabled appearance when there are validation errors
+      ZulipWebUiKitButton(label: zulipLocalizations.composeBoxBannerButtonSave,
+        attention: ZulipWebUiKitButtonAttention.high,
+        onPressed: () {
```

Comment:
nit: this callback is about 20 lines; once it's more than a small handful of lines, I think it's a bit easier to read as a separate named function, so here a private method on this class.